File: PSB/Api/Request/RestoreFromKeyRequest.cs
using System.Text.Json.Serialization;

namespace PSB.Api.Request
{
    public class RestoreFromKeyRequest(string login, string key, string password, string passwordConfirmation, bool logout)
    {
        [JsonPropertyName("login")] public string Login { get; set; } = login;
        [JsonPropertyName("key")] public string Key { get; set; } = key;
        [JsonPropertyName("new_password")] public string NewPassword { get; set; } = password;
        [JsonPropertyName("new_password_confirmation")] public string NewPasswordConfirmation { get; set; } = passwordConfirmation;
        [JsonPropertyName("logout")] public bool Logout { get; set; } = logout;
    }
}

File: Laravel/app/Http/Requests/Api/Auth/RestoreFromKeyRequest.php
<?php

namespace App\Http\Requests\Api\Auth;

use App\Http\Requests\ApiRequest;

class RestoreFromKeyRequest extends ApiRequest
{
    public function rules(): array
    {
        return [
            'login' => 'required|string|exists:users,login',
            'key' => 'required|string',
            'new_password' => 'required|string|min:6|confirmed',
            'new_password_confirmation' => 'required|string|min:6',
            'logout' => 'boolean',
        ];
    }
}
